actions/lint-pr-title/src/main.ts

```diff
@@ -0,0 +1,308 @@
+  switch (strings.length) {
+    case 0:
+      return "";
+    case 1:
+      return strings[0];
+    case 2:
+      return strings.join(" or ");
+    default:
+      return `${strings.slice(0, -1).join(", ")}, or ${
+        strings[strings.length - 1]
+      }`;
+  }
+}
+
+/**
+ * Error thrown when the commit message is invalid.
+ */
+export class LintError extends Error {
+  constructor(public readonly report: LintResult) {
+    super(format(report));
+  }
+}
+
+/**
+ * Error thrown when the event type is not one of the types this action can
+ * handle.
+ */
+export class WrongEventTypeError extends Error {
+  private static HANDLED_TARGETS = [
+    "merge_group",
+    "pull_request_target",
+    "pull_request",
+  ];
+
+  constructor(public readonly event: string) {
+    const message = `This action can only be run on ${joinStringWithOr(
+      WrongEventTypeError.HANDLED_TARGETS.map((target) => `\`${target}\``),
+    )} events. Got event \`${event}\`.`;
+
+    super(message);
+  }
+}
+
+export interface LintResult {
+  valid: boolean;
+  errorCount: number;
+  warningCount: number;
+  results: (LintOutcome & { sha?: string })[];
+}
+
+/**
+ * Lint the messages against the provided rules. If the message is invalid, an
+ * error is thrown.
+ *
+ * @param messages - The list of commit or PR messages to lint.
+ * @param config - The config object containing the rules to lint against.
+ * @throws {LintError} If the title is invalid.
+ */
+export async function lint(
+  messages: { sha?: string; message: string }[],
+  { parserPreset, plugins, ignores, defaultIgnores, rules }: QualifiedConfig,
+): Promise<LintResult> {
+  if (messages.length === 0) {
+    return {
+      valid: true,
+      errorCount: 0,
+      warningCount: 0,
+      results: [],
+    };
+  }
+
+  const opts = {
+    parserOpts: parserPreset?.parserOpts ?? {},
+    plugins,
+    ignores: ignores ?? [],
+    defaultIgnores: defaultIgnores ?? true,
+  };
+
+  const results = await Promise.all(
+    messages.map(async ({ sha, message }) => {
+      console.log(`Linting message: ${message}`);
+      return { sha, ...(await commitLint(message, rules, opts)) };
+    }),
+  );
+
+  const report = results.reduce(
+    (info, result) => {
+      info.valid = result.valid ? info.valid : false;
+      info.errorCount += result.errors.length;
+      info.warningCount += result.warnings.length;
+      info.results.push(result);
+
+      return info;
+    },
+    {
+      valid: true,
+      errorCount: 0,
+      warningCount: 0,
+      results: <LintOutcome[]>[],
+    },
+  );
+
+  return report;
+}
+
+/**
+ * Handle the pull request event. The title of the pull request is linted
+ * against the provided rules. If the title is invalid, an error is thrown.
+ *
+ * @param prCtx - The pull request context.
+ * @param octokit - The authenticated Octokit instance.
+ * @param config - The configuration object.
+ * @throws If the title is invalid.
+ */
+export async function handlePullRequest(
```

Comment:
I believe the docs comment for this function says it should have 3 parameters, but I only see 2: WebHookPayload and Config.

The Octokit parameter is missing, maybe it's not used and need to be removed from the comment?

Reply:
Thanks yeah, I deleted it at some point, will update 👍